specs/logging_implementation.adoc

```diff
@@ -0,0 +1,174 @@
+  and it should be able to deliver them to either syslog, log files, alert
+  sockets etc. depending upon the metadata and configuration.
+* Syslog must be the default logging configuration. Logging to individual
+  configuration files should be thoroughly discouraged.
+
+== Use Cases
+
+Logs and alerts need to be treated as messages.
+
+== Proposed change
+* Create a new class called "Message" in common with attributes name, type,
+  message, priority(empty for log), alert_id ,log_level(empty for alert),
+  function_name, line_number.
+  (Logging will take line number and function name from where it is called, Here
+  the problem is we are logging the message from common so it takes a function name
+  and line number from common only. So it makes debugging complex. To avoid this
+  we have to find the function name, line number, file name and append this data
+  with message manually.
+```
+from inspect import getframeinfo, stack
+caller = getframeinfo(stack()[1][0])
+caller.filename, caller.lineno, caller.function
+(These are built in packages in python)
+```
+* Using name retrieve the particular logging handler.
+* Create a new function for Message class and process the object values.
+```
+1. If the type is Alert then create a new uuid and assign the uuid to alert. After
+   that convert the object into json, If the json is created successfully then store
+   json into log file as warning and put the json in node_agent socket also.
+2. If the type is Log then process the object and store json in log file based on
+   log level.
+```
+* Import the Message class in all the modules and Change log statements to
+  populate the instance of message class and pass the object to function.
+* Expected output format for log and alert in log file:
+```
+Format: "%(asctime)s - %(levelname)s - %(message)s"
+Message: "%(filename)s - %(function)s - %(lineno)s - %(message)s"
+```
+* Expected output format for the alert in node_agent socket:
+```
+{
+   "alert_id" : "uuid",
+   "priority" : "string",
+   "message"  : "string"
+}
+```
+
+== Alternatives
+
+* None
+
+== Data model impact
+
+None
+
+== Impacted modules
+
+=== Tendrl/common impact
+
+* Create a new class called "Message" in common with attributes name, type,
+  message, priority(empty for log), alert_id ,log_level(empty for alert),
+  function_name, line_number.
+* Using name retrieve the particular logging handler.
+* Create a new function for Message class and process the object values.
+* If the type is Alert then create a new uuid and assign the uuid to alert. After
+  that convert the object into json, If the json is created successfully then store
+  json into log file and put the json in node_agent socket also.
+* If the type is Log then process the object and store json in log file based on
+  log level.
+
+=== Tendrl/node_agent impact
+
+* Import the Message class in all the modules and Change log statements to
```

Comment:
I feel we should add the framework in one PR and do the mass change in different PR. Makes life easier and doesnt halt everybody else :)